commit 2d66928d6180d3422a25e63161b93816574cfa31
Author: leo.singer@LIGO.ORG <leo.singer@LIGO.ORG@ac820499-f4a4-4ea9-b4e2-a7172f5148fc>
Date:   Wed Sep 11 21:05:20 2013 +0000

    Check default include path for jni.h before resorting to AX_JNI_INCLUDE_DIR
    
    git-svn-id: https://svn.ligo.caltech.edu/svn/nds2/trunk@1155 ac820499-f4a4-4ea9-b4e2-a7172f5148fc

diff --git configure.ac configure.ac
index e5c388f..4399425 100644
--- configure.ac
+++ configure.ac
@@ -496,12 +496,17 @@ dnl java
 AS_IF(
 	[test "x$enable_swig_java" != xno],
 	[
-		AX_JNI_INCLUDE_DIR
-		for JNI_INCLUDE_DIR in $JNI_INCLUDE_DIRS
-		do
-			JNI_CPPFLAGS="$JNI_CPPFLAGS -I$JNI_INCLUDE_DIR"
-		done
-		AC_SUBST([JNI_CPPFLAGS], [$JNI_CPPFLAGS])
+		dnl Check to see if jni.h is in the default include path.
+		dnl If it isn't, try to guess where it is relative to the Java compiler
+		dnl using AX_JNI_INCLUDE_DIR.
+		AC_CHECK_HEADER([jni.h], , [
+			AX_JNI_INCLUDE_DIR
+			for JNI_INCLUDE_DIR in $JNI_INCLUDE_DIRS
+			do
+				JNI_CPPFLAGS="$JNI_CPPFLAGS -I$JNI_INCLUDE_DIR"
+			done
+			AC_SUBST([JNI_CPPFLAGS], [$JNI_CPPFLAGS])
+		])
 		case $target_os in
 		darwin*)
 		  JAVA_JNI_LIB_PREFIX="lib"
